Design.md

```diff
@@ -227,8 +229,26 @@ This API will enable or disable user on Vault.
 
 
 ### Create S3 Credential
-This API creates S3 credential for the user.
-1. `generate-access-key` api will be called using assumed role credentials.
+This API creates S3 credentials for the user.
+1. The Generate Access Key API will be called using assumed role credentials.
+1. Encrypt the secret key (For more, see [SecretKey-Encryption-Strategy](#SecretKey-Encryption-Strategy)). Store the encrypted secret key on Redis Sentinel in the hash named "osis:s3credentials", with the hash key formatted as `<Username>__<AccessKeyID>`.
+1. Log the access key ID _(Log location and tools for logging are yet to be decided)_
+
+**Important Notes:**
+* Redis Sentinel is subject to an ongoing rolling deployment.
+* If Redis Sentinel fails or crashes, the vCloud Director's OSE service must be restarted with the following command:
+    ```shell
+    $ ose service restart
+    ```
```

Comment:
Out of curiosity: what type of failure are we expecting on redis-sentinel which will require OSE to be restarted?

If redis-sentinel has quorum then there shouldn't be a need to restarting OSE service.

Reply:
That is true Redis sentinel provides automatic failover and is highly available. This is a scenario for a rainy day when Redis Master and all it's Replicas fail (As we won't have any source of truth other than Redis)

Reply:
I would suggest then mentioning it explicitly. 
What we mean here by fail/crashes? (Maybe loss of quorum or if redis sentinel cluster is not available)

Reply:
Fixed it in e87f4cb37a396aa2237d83bf61f623d6fb24ccc4